railties/lib/rails/commands/notes/notes_command.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "rails/source_annotation_extractor"
+
+module Rails
+  module Command
+    class NotesCommand < Base # :nodoc:
+      class_option :annotations, aliases: "-a", desc: "Filter by specific annotations, e.g. Foobar TODO", type: :array, default: %w(OPTIMIZE FIXME TODO)
+
+      def perform(*)
+        require_application_and_environment!
+
+        display_deprecation_warning
+        display_annotations
+      end
+
+      private
+        def display_annotations
+          annotations = options[:annotations]
+          tag = (annotations.length > 1)
+
+          say Rails::SourceAnnotationExtractor.enumerate annotations.join("|"), tag: tag, dirs: directories
+        end
+
+        def directories
+          Rails::SourceAnnotationExtractor::Annotation.directories + (source_annotation_directories || "").split(",")
+        end
+
+        def display_deprecation_warning
+          return unless source_annotation_directories
+          say ActiveSupport::Deprecation.warn("`SOURCE_ANNOTATION_DIRECTORIES` will be deprecated in Rails 6.1. You can add default directories by using config.annotations.register_directories instead.")
+          say "\n"
+        end
+
+        def source_annotation_directories
+          ENV["SOURCE_ANNOTATION_DIRECTORIES"]
+        end
```

Comment:
I'd move the `split` to here (and replace `|| ""` with `to_s`).

It makes sense that a `ENV["SOURCE_ANNOTATION_DIRECTORIES"]` returns a string or nil, but it doesn't make sense that a Ruby method named such does. That would be an empty array at worst.

Then `display_deprecation_warning` can do:

```ruby
if source_annotation_directories.empty?
  say …
end
```